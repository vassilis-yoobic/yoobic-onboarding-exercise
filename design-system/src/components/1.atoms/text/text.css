.text {
  text-decoration: none;
  display: inline-block;
}
/*# sourceMappingURL=text.css.map */