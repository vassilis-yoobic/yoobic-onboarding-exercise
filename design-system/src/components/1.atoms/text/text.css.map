{
    "version": 3,
    "mappings": "AAAA,AAAA,KAAK,CAAC;EACJ,eAAe,EAAE,IAAI;EACrB,OAAO,EAAE,YAAY;CACtB",
    "sources": [
        "text.scss"
    ],
    "names": [],
    "file": "text.css"
}