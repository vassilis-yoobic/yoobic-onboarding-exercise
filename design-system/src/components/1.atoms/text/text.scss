.text {
  text-decoration: none;
  display: inline-block;
}

