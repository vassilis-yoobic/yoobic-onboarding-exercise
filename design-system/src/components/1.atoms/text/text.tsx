import { Component, ComponentInterface, h, Host, Prop } from '@stencil/core';

@Component({
  tag: 'cu-text',
  styleUrl: 'text.css',
})

export class CustomText implements ComponentInterface {
  @Prop() content!: string;
  @Prop() size?: number = 14;
  @Prop() color?: string = '#03060b';



  render() {
    return (
      <Host>
        <div class="text" style={{color: this.color, fontSize: this.size + 'px' }}>
        {this.content}
        </div>
      </Host>);
  }
}
