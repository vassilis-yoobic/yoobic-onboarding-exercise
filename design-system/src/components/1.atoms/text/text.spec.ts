import { newSpecPage } from "@stencil/core/testing";

import { CustomText } from "./text";

describe('1.atoms/text', () => {
  describe('rendering', () => {
    let page;

    beforeEach(async () => {
      page = await newSpecPage({
        components: [CustomText],
        html: '<cu-text>Test</cu-text>'
      })
    });
    describe('props', () => {
      it('should render', async () => {
        expect(page.root).toMatchSnapshot();
      })
      it('should render with different color', async () => {
        page.root.color = '#123456';
        await page.waitForChanges();
        expect(page.root).toMatchSnapshot();
      })
      it('should render with different font size', async () => {
        page.root.size = '18px';
        await page.waitForChanges();
        expect(page.root).toMatchSnapshot();
      })
    })
  })
})
