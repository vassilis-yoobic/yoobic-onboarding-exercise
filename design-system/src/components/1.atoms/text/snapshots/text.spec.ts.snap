exports[`1.atoms/text rendering props should render`] = `
<cu-text>
  <div class="text" style="color:#000; font-size:14px;">
    Some Text
  </div>
</cu-text>
`;

exports[`1.atoms/text rendering props should render with different color`] = `
<cu-text>
  <div class="text" style="color:#123456;">
    Some Text
  </div>
</cu-text>
`;

exports[`1.atoms/text rendering props should render with different font size`] = `
<cu-text>
  <div class="text" style="font-size:18px;">
    Some Text
  </div>
</cu-text>
`;
